source/MaterialXCore/Document.cpp

```diff
@@ -315,6 +320,12 @@ void Document::upgradeVersion()
         }
         minorVersion = 23;
     }
+    if (majorVersion == desiredMajorVersion &&
+        minorVersion == desiredMinorVersion)
+    {
+        setVersionString(makeVersionString(majorVersion, minorVersion));
+        return;
+    }
```

Comment:
Since this feature really exists to allow clients to request _future_ MaterialX versions, maybe it would be better to explicitly add a "applyFutureUpgrades" flag instead of allowing the user to request arbitrary past versions.  This would remove a good deal of boilerplate code from the upgradeVersion() method, and would make it more clear which choice is relevant to the client.  By default, documents would upgrade to the current major and minor version, and the new flag would allow them to apply all future upgrades that are currently supported (e.g. converting material elements to nodes for 1.38).

Reply:
Agreed. This does make the upgrade path simpler to manage as well, and it is not likely that a partial upgrade path would work due to other dependent changes.